### PR TITLE
doc: Clarify includeWatching for fundrawtransaction

### PR DESCRIPTION
**Pull request by stevenroose**
Might be sufficient to solve https://github.com/bitcoin/bitcoin/issues/16396, https://github.com/bitcoin/bitcoin/issues/7879 and https://github.com/bitcoin/bitcoin/issues/14405.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3146,7 +3146,8 @@ static UniValue fundrawtransaction(const JSONRPCRequest& request)
                             {"changeAddress", RPCArg::Type::STR, /* default */ "pool address", "The bitcoin address to receive the change"},
                             {"changePosition", RPCArg::Type::NUM, /* default */ "random", "The index of the change output"},
                             {"change_type", RPCArg::Type::STR, /* default */ "set by -changetype", "The output type to use. Only valid if changeAddress is not specified. Options are \"legacy\", \"p2sh-segwit\", and \"bech32\"."},
-                            {"includeWatching", RPCArg::Type::BOOL, /* default */ "false", "Also select inputs which are watch only"},
+                            {"includeWatching", RPCArg::Type::BOOL, /* default */ "false", "Also select inputs which are watch only.\n"
+                            "                              Only solvable inputs can be used."},
```
Comment (MarcoFalke) [inline src/wallet/rpcwallet.cpp]: style-nit: Whitspace should be trimmed from strings later anyway:

```suggestion
                                                          "Only solvable inputs can be used."},
```

**Comment by jonasschnelli** [comment]
I think this change is helpful.
Though giving a short instruction how one can import a solveble-watch-only would probably be even better.
An alternative would be returning a more precise error (rather than just "insufficient funds").

**Comment by stevenroose** [comment]
Pushed an update explaining how to get solvable inputs.

**Comment by jb55** [comment]
Concept ACK, although I think a better error message is the right approach here. Insufficient funds is a bit vague.

I believe the same applies to walletcreatefundedpsbt?

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3146,7 +3146,9 @@ static UniValue fundrawtransaction(const JSONRPCRequest& request)
                             {"changeAddress", RPCArg::Type::STR, /* default */ "pool address", "The bitcoin address to receive the change"},
                             {"changePosition", RPCArg::Type::NUM, /* default */ "random", "The index of the change output"},
                             {"change_type", RPCArg::Type::STR, /* default */ "set by -changetype", "The output type to use. Only valid if changeAddress is not specified. Options are \"legacy\", \"p2sh-segwit\", and \"bech32\"."},
-                            {"includeWatching", RPCArg::Type::BOOL, /* default */ "false", "Also select inputs which are watch only"},
+                            {"includeWatching", RPCArg::Type::BOOL, /* default */ "false", "Also select inputs which are watch only. Only solvable inputs can be used.\n"
+                                                          "Watch-only destinations are solvable if the public key was imported, \n"
+                                                          "f.e. with 'importpubkey' or 'importmulti' with the 'pubkeys' or 'desc' field."},
```
Comment (meshcollider) [inline src/wallet/rpcwallet.cpp]: nit: `f.e.` -> `e.g.`

Comment (stevenroose) [reply src/wallet/rpcwallet.cpp]: fixed

**Comment by meshcollider** [comment]
The "public key" solvability is really only true for single-key destinations like P2PK, P2PKH and P2WPKH. For script-hash type outputs, you need to know the script, etc.

**Comment by stevenroose** [comment]
> The "public key" solvability is really only true for single-key destinations like P2PK, P2PKH and P2WPKH. For script-hash type outputs, you need to know the script, etc.

I'll add `and/or scriptPubkey` and rebase.

**Review by meshcollider — COMMENTED** [review]
I'm not sure if the term "output script" is standard, solvability for P2SH for example requires the wallet to know the embedded script not the scriptPubKey. But this is okay I guess, I don't want to bikeshed when the meaning is clear.

**Comment by stevenroose** [comment]
@meshcollider I think the main purpose of the second part of the explanation is that there are other requirements for solvability that include pubkeys and scripts. I'm not sure how wide-spread it is, but I generally use input script and output script as synonyms for resp. scriptSig's and scriptPubKey's. And sure in the p2sh case there's kinda two output scripts going on of which one is just a hashed version of the other.

**Comment by Sjors** [comment]
ACK 8003104. This will always be confusing, but at least it gives a bunch more clues for the user to google.
Schermafbeelding 2019-09-27 om 19 57 16

**Pull request merged**